### PR TITLE
History

### LINKED ISSUE #24: Up key does not propose the last query, but the latest command
If I type:
```console
vm list
? [].name
? [].id
```
I expect the up key to show "? [].id", not "vm list"

image

### PR DESCRIPTION
**Pull request by oakeyc**
fixes #24

### CODE REVIEW

File: azclishell/az_completer.py [hunk 1]
```diff
@@ -125,18 +125,18 @@ def get_completions(self, document, complete_event):
...
-        for cmd in self.gen_cmd_completions(text):
+        for cmd in sorted(list(self.gen_cmd_completions(text)), key=lambda a: a.text):
             yield cmd
 
-        for val in self.gen_dynamic_completions(text):
+        for val in sorted(list(self.gen_dynamic_completions(text)), key=lambda a: a.text,):
             yield val
 
-        for param in self.gen_global_param_completions(text):
+        for param in sorted(list(self.gen_global_param_completions(text)), key=lambda a: a.text,):
```
Comment (derekbekoe) [inline azclishell/az_completer.py]: This should be in a helper method that you call.
Something like `sorted_completions(...)`

File: azclishell/configuration.py [hunk 2]
```diff
@@ -22,8 +22,8 @@
     SELECT_SYMBOL['query'] + "[path]" : "query previous command using jmespath syntax",
     "[cmd] " + SELECT_SYMBOL['example'] + " [num]" : "do a step by step tutorial of example",
     SELECT_SYMBOL['exit_code'] : "get the exit code of the previous command",
-    SELECT_SYMBOL['default'] : "default a scope",
-    SELECT_SYMBOL['undefault'] : "undefault a scope",
+    SELECT_SYMBOL['default'] + '[cmd]' : "set a scope",
```
Comment (derekbekoe) [inline azclishell/configuration.py]: These keys should be changed as now the meaning as changed.
`set-scope` `unset-scope` maybe.
Defaults mean something else in the CLI.

**Review by derekbekoe — CHANGES_REQUESTED** [review]

**Comment by oakeyc** [comment]
changes made

File: azclishell/__init__.py [hunk 3]
```diff
@@ -2,4 +2,4 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
-__version__ = '0.1.1a27'
+__version__ = '0.1.1a31'
```
Comment (derekbekoe) [inline azclishell/__init__.py:5]: This version does not match the one in setup.py

**Review by derekbekoe — CHANGES_REQUESTED** [review]
One more thing.
The version no.s don't match up.

Would suggest not changing the version no. in PRs otherwise you'll get version mismatches depending on the order PRs are merged.
Only change version no.s before a release in a release PR.

**Review by derekbekoe — APPROVED** [review]

**Pull request merged**